torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -2007,11 +2027,14 @@ void ProcessGroupNCCL::broadcastDumpSignal() {
     shouldDump_.store(true);
     // Give time for dumping before throwing exception
     auto start = std::chrono::steady_clock::now();
+    // Give 2 * waitTimeoutDumpInMilSec_ to dump the flight recorder.
+    // We try capturing with stack traces first, and if it fails, we try without
+    // stack traces.
     auto status = promiseFlightRecorderDump_.get_future().wait_for(
-        std::chrono::milliseconds(waitTimeoutDumpInMilSec_));
+        std::chrono::milliseconds(2 * waitTimeoutDumpInMilSec_));
```

Comment:
nit: `waitTimeoutDumpInMilSec_` is not used elsewhere, maybe we can directly bump its value by 2x? (easier maintenance I guess)

Reply:
So the watchdog needs to give 2 mins for heartbeat thread to try to dump (1 min for with stack trace and if this fails, 1 min without stack trace).